Debian/bin/lib/Cyrus/Docker/Command/build.pm

```diff
@@ -22,10 +29,22 @@ sub opt_spec {
   );
 }
 
+sub recompile ($self, $opt, $args) {
+  say "Recompiling...";
+
+  run(qw( make -j 8 ));
+  run(qw( make -j 8 check ));
+  run(qw( sudo make install ));
+
+  system('/usr/cyrus/bin/cyr_info', 'version');
+}
+
 sub execute ($self, $opt, $args) {
   my $root = $self->app->repo_root;
   chdir $root or die "can't chdir to $root: $!";
 
+  return $self->recompile($opt, $args) if $opt->recompile;
```

Comment:
I think you should turn this inside out.

1. delete `->recompile`
2. move everything between `chdir` and the code that `recompile` was duplicating into `->configure`
3. then call `$self->configure unless $opt->recompile`

This eliminate duplication and better encapsulates that one hunk of code as the config code.

This may suggest more work around `make check`, but maybe latter commits make that smoother.